### ISSUE TITLE
Fix support for ValidatingAdmissionWebhook and MutatingAdmissionWebhook in custom API servers

### ISSUE THREAD

**Issue by nilebox** [kubernetes/kubernetes#60963]
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug
/sig api-machinery

**What happened**:
The `ValidatingAdmissionWebhook` and `MutatingAdmissionWebhook` admission controllers can be enabled in custom API servers, but they are actually broken.
To fix that in each API server we have to import `k8s.io/kubernetes/pkg/apis/admission/install` and explicitly call the corresponding `Install()` method to add necessary types to the Scheme. In other words, we have to either add a dependency on `k8s.io/kubernetes`, or copy the source code that installs the `v1beta1.AdmissionReview` type to the Scheme.

**What you expected to happen**:
Enabling ValidatingAdmissionWebhook should work without writing any extra code at all (just passing a flag `--admission-control=ValidatingAdmissionWebhook` should be enough).

**Proposed solution**:
1. Move `k8s.io/kubernetes/pkg/apis/admission/install` package from the `k8s.io/kubernetes` to the `k8s.io/apiserver` repo (or create copy of this code in `k8s.io/apiserver`)
2. Invoke the `Install()` method adding the `AdmissionReview` type to the Scheme in the API server by default.

**How to reproduce it (as minimally and precisely as possible)**:
If we just enable `ValidatingAdmissionWebhook` in the `sample-apiserver`, the server will startup successfully. But if we try creating a sample resource, we'll get the following error:
```
Error from server (InternalError): error when creating "artifacts/flunders/01-flunder.yaml": Internal error occurred: failed calling admission webhook "warbles.k8s.io": no kind is registered for the type v1beta1.AdmissionReview
```
In other words, we have to register the type `v1beta1.AdmissionReview` in order to make this webhook work.
See PR https://github.com/kubernetes/sample-apiserver/pull/21 for amount of work that is needed to enable ValidatingAdmissionWebhook and MutatingAdmissionWebhook admission controllers in custom API server.

**Environment**:
- Kubernetes version (use `kubectl version`): 1.9.2

**Comment by nilebox** [comment]
@deads2k @caesarxuchao This issue is a replacement for #60924 clarifying the actual issue with enabling `ValidatingAdmissionWebhook` and `MutatingAdmissionWebhook` in custom API servers.

**Comment by nilebox** [comment]
Apparently, the `AdmissionReview` type, as well its registration, is duplicated in two places:
https://github.com/kubernetes/api/blob/master/admission/v1beta1/register.go
https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/admission/register.go
Not sure what's the reasoning for this, but to fix the issue we really just need to copy the https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/admission/install/install.go file and reference `AdmissionReview` type defined in `k8s.io/api`.
And invoke this `admissioninstall.Install()` method from somewhere in `k8s.io/apiserver`.

**Comment by nilebox** [comment]
There is https://github.com/kubernetes/apiserver/blob/master/pkg/apis/audit/install/install.go, so for consistency should probably copy the `k8s.io/kubernetes/pkg/apis/admission/install/install.go`  to `k8s.io/apiserver/blob/master/pkg/apis/admission/install/`

**Comment by sttts** [comment]
`pkg/apis` holds the internal types corresponding to the external types in `k8s.io/api`.  The former is not to be used by anyone outside of the kube-apiserver. The install.go is for kube-apiserver only for the same reason.

If we are lacking the registration of the types, we have to do that in a webhook local (invisible for sample-apiserver code) scheme. Also the admission webhooks should use versioned types. Do they?

**Comment by nilebox** [comment]
Note: this bug was discovered as part of https://github.com/kubernetes-incubator/service-catalog/issues/1812 and https://github.com/kubernetes-incubator/service-catalog/pull/1811, and we need this fix in 1.10 at least (and preferably in 1.9.2 too).

**Comment by tpepper** [comment]
@nilebox do you see this as a blocking issue then for 1.10?  Or something you'll look to cherry pick later in 1.10.x?

**Comment by sttts** [comment]
@tpepper it only affect 3rdparty apiservers. So having it as a bugfix release is good enough. This patch should not delay kube. But we have a PR (and another one from @caesarxuchao), without much risk.

**Comment by nilebox** [comment]
@sttts should this issue remain opened until the bugfix is backported to 1.10 and 1.9?
If not, how do I otherwise track if it was cherrypicked?

**Comment by caesarxuchao** [comment]
1.10 is still doing branchff, so the PR will appear in 1.10.
I sent #61066 as the cherrypick to 1.9.
